src/Build/BackEnd/Components/RequestBuilder/AssemblyLoadsTracker.cs

```diff
@@ -157,11 +157,14 @@ private void CurrentDomainOnAssemblyLoad(object? sender, AssemblyLoadEventArgs a
                 : $"{_appDomain.Id}|{_appDomain.FriendlyName}";
 
 
-            AssemblyLoadBuildEventArgs buildArgs = new(_context, _initiator, assemblyName, assemblyPath, mvid, appDomainDescriptor)
+            AssemblyLoadBuildEventArgs buildArgs = new(_context, _initiator, assemblyName, assemblyPath, mvid, appDomainDescriptor);
+
+            // Fix #8816 - when LoggingContext does not have BuildEventContext it is unable to log anything
+            if (_loggingContext?.BuildEventContext != null)
             {
-                BuildEventContext = _loggingContext?.BuildEventContext ?? BuildEventContext.Invalid
-            };
-            _loggingContext?.LogBuildEvent(buildArgs);
+                buildArgs.BuildEventContext = _loggingContext.BuildEventContext;
+                _loggingContext.LogBuildEvent(buildArgs);
+            }
             _loggingService?.LogBuildEvent(buildArgs);
```

Comment:
do you need to remove this line now?

Reply:
It has not been removed, only moved to `if` at line 166

Reply:
so, do we need both LoggingContext.LogBuildEvent() as well as LoggingService.LogBuildEvent()? sorry I don't know what exactly these two methods do and what is the difference.

Reply:
Only one of those is set. If we have LoggingContext than we use it, in cases where LoggingContext is unknown, for example logger intialization, we do use LoggingService instead.